components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -484,7 +499,9 @@ class SitePermissionsFeature(
         return false
     }
 
-    private val PermissionRequest.host get() = uri?.toUri()?.host ?: ""
+    private fun PermissionRequest.getHost(session: Session) = (uri?.toUri()?.host
+            ?: "").ifEmpty { session.url.toUri().host ?: "" }
```

Comment:
nit: Using an empty string only to then check if it is an empty string seems unnecessary. Afaik we can chain them via the nullable check? `uri?.toUri()?.host
    ?: session.url.toUri().host ?: ""`

Reply:
:sweat_smile: 
Thanks for catching this!

Reply:
The only thing is bit hard to read, for the nested checks 😅, I added it as it a bit clear, but I guess it's matter of taste